### ISSUE TITLE
ambiguous types

### ISSUE THREAD

**Issue by thomvet** [martinkosch/open62541.jl#11]
Currently, there are a number of ambiguous types (several UA types have the same julia type), namely:
```
types_ambiguous_ignorelist = [
    :UA_Duration, #Float64
    :UA_ByteString, #UA_String
    :UA_XmlElement, #UA_String
    :UA_LocaleId, #UA_String
    :UA_DateTime, #Int64
    :UA_UtcTime, #Int64
    :UA_StatusCode, #UInt32
]
```
I am wondering whether this is a problem when we are defining UA_XXX_delete methods that look up the correct UA type within UA_TYPES_PTRS based on the julia type.

For the string types, it would for example be quite easy to make them distinct (just copy the UA_String struct definition); for the number types, it would be slightly more involved, but I think it could in principle be done using AbstractNumbers.jl.

**Comment by martinkosch** [comment]
Yes, this currently definitely leads to a problem as the user is required to add the correct `type_ptr`argument to the functions. AbstractNumbers.jl indeed looks promising for a nice solution.
